src/main/java/org/springframework/data/elasticsearch/core/DefaultResultMapper.java

```diff
@@ -206,4 +209,16 @@ private String buildJSONFromFields(Collection<DocumentField> values) {
 			}
 		}
 	}
+
+	private <T> void setPersistentEntityScore(T result, float score, Class<T> clazz) {
+		if (mappingContext != null && clazz.isAnnotationPresent(Document.class)) {
```

Comment:
Do we need that guard? Shouldn't the constructor of `DefaultResultMapper` already make sure that it's never `null`? What's the second guard forI'd vote to avoid any Java-native reflection in the context of actually working with the metamodel.